### ISSUE TITLE
Easy way to add a let's encrypt cert

### ISSUE THREAD

**Issue by robetus** [inovector/mixpost#34]
Is there an easy way to add a let's encrypt certificate to the compose.yml file maybe using certbot? I see that mixpost is using apache2 how can I add a let's encrypt cert for mixpost url?

**Comment by lao9s** [comment]
Hi @robetus ,
Mixpost docker doesn't support let's encrypt or any SSL certificates generation. You have to deal with it yourself.

There is a very useful tool: https://nginxproxymanager.com/
It's easy to install with docker and configuring proxy + SSL certificates.

**Comment by robetus** [comment]
Thank you @lao9s but I can't get the ports right. Nginx Proxy Manager needs port 80, 81, 443 and mixpost is defaulted to port 80. I tried to change mixpost a different port in the compose yaml file but then the cert doesn't secure everything. NPM gets a cert for mixpost just fine but it states that the content is not secure when I change the mixpost port from the default port 80.

**Comment by lao9s** [comment]
- What has NPM to do with certificates?
- How did you generate the certificates?
- Are you accessing the Mixpost dashboard with a real domain or an IP address?

**Comment by robetus** [comment]
> Hi @robetus , Mixpost docker doesn't support let's encrypt or any SSL certificates generation. You have to deal with it yourself.
> 
> There is a very useful tool: https://nginxproxymanager.com/ It's easy to install with docker and configuring proxy + SSL certificates.

I used NPM at your suggestion, I generated the certificates with npm. I am using real public IP addresses with real public domains, not internal IP addresses.

**Comment by saschafoerster** [comment]
Hi @robetus I also think that Proxies and Certs shouldn't be part of Mixpost, as there are many ways to achieve this, depending on the servers and environments you are using. 

I also think you have a misunderstanding: 
- there is the Nginx Proxy Manager: https://nginxproxymanager.com
- and then there is the Node Package Manager, mostly called NPM: https://www.npmjs.com

Still I want to give you a little help. I am using nginx as a proxy, certbot for my certs. 

So first find out, how to use certbot on your server and create certificates with let's encrypt. 
Secondly, learn how to use Nginx Proxy Manager or Nginx. 
I added this to the configuration of my nginx: 

```
server {
    server_name mixpost.mydomain.tld; # managed by Certbot
    location / {
            proxy_pass http://127.0.0.1:8080;
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
    }
    listen [::]:443 ssl; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/mixpost.mydomain.tld/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/mixpost.mydomain.tld/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}
server {
    if ($host = mixpost.mydomain.tld) {
        return 301 https://$host$request_uri;
    } # managed by Certbot
	listen 80 ;
	listen [::]:80 ;
    server_name mixpost.mydomain.tld;
    return 404; # managed by Certbot
}

```

In short explained: Nginx is listening to port 80 and 443. If someone asks for mixpost.mydomain.tld, then it is proxied to port 8080 on the localhost 127.0.0.1

So, now you have to make Mixpost listen to :8080 in docker compose: 

```
services:
    mixpost:
        image: inovector/mixpost:latest
...
        ports:
            - 127.0.0.1:8080:80
```

127.0.0.1:8080 is inbound, :80 shows, where in the container itself the server is listening. 

Hope that helps.

**Comment by robetus** [comment]
Hi @saschafoerster I wasn't getting the two confused I was trying to use Nginx Proxy Manager (NPM) to install a cert for mixpost on the same server running both NPM and mixpost. I figured that installing nginx and certbot was the easiest way to do it but I wanted to give NPM a try, it's a really awesome tool. Using nginx and certbot seems pretty easy as you have shown.

**Comment by robetus** [comment]
I achieved this with portainer and Nginx Proxy Manager, works amazingly.
